framework/src/modules/interoperability/base_cross_chain_update_command.ts

```diff
@@ -65,6 +65,15 @@ export abstract class BaseCrossChainUpdateCommand<
 		await this.internalMethod.verifyCertificateSignature(context, params);
 
 		if (!isInboxUpdateEmpty(inboxUpdate)) {
+			// This check is expensive. Therefore, it is done in the execute step instead of the verify
+			// step. Otherwise, a malicious relayer could spam the transaction pool with computationally
+			// costly CCU verifications without paying fees.
+			try {
+				await this.internalMethod.verifyPartnerChainOutboxRoot(context, params);
```

Comment:
tests are missing that 

1. this function is called
2. `executeCommon` fails if `verifyPartnerChainOutboxRoot` fails